homeassistant/components/climate/nest.py

```diff
@@ -37,11 +38,14 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
 
     temp_unit = hass.config.units.temperature_unit
 
-    add_devices(
-        [NestThermostat(structure, device, temp_unit)
-         for structure, device in hass.data[DATA_NEST].thermostats()],
-        True
-    )
+    all_devices = [NestThermostat(structure, device, temp_unit)
+                   for structure, device in hass.data[DATA_NEST].thermostats()]
+
+    add_devices(all_devices, True)
+
+    for device in all_devices:
+        async_dispatcher_connect(hass, SIGNAL_NEST_UPDATE,
```

Comment:
This should be done inside the entity coroutine method `async_added_to_hass`. Otherwise there might be an error if the entity state update methods are called before the entity has been added to home assistant.